### PR TITLE
Add 'Nothing selected' when a chart is empty

### PR DESCRIPTION
**Pull request by MatthewKhouzam**
I can't find the issue that it was solving. It's something like "show data by default"

### CODE REVIEW

**Comment by MatthewKhouzam** [comment]
Before a chart would be empty if there was no data in it. Now there's a warning. 
nothing-selected

File: packages/react-components/src/components/xy-output-component.tsx [hunk 1]
```diff
@@ -232,6 +232,16 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
                 }
             </React.Fragment>;
         }
+        if (this.state.outputStatus === ResponseStatus.COMPLETED && this.state.checkedSeries.length === 0 ) {
```
Comment (MatthewKhouzam) [inline packages/react-components/src/components/xy-output-component.tsx:235]: How about !!this.state.xyData? to handle those views that have total?

**Review by MatthewKhouzam — COMMENTED** [review]
Needs an issue number

**Comment by ebugden** [comment]
> I can't find the issue that it was solving. It's something like "show data by default"

That's my fault! I never got around to creating the issue. Here you go: #604

**Review by ebugden — CHANGES_REQUESTED** [review]
Requested changes:

- [x] Simplify message. Suggestion: "Check a box to see analysis results"

I still have a strong preference for just checking the first box by default (for the views that don't have an obvious default) instead of showing a message: 
* In my mind, seeing a checked box shows that checking boxes is possible and where they can be selected in the UI. A checked box clearly tutorializes the action way more explicitly than a message. 
* I also think this better tutorialization outweighs the fact that if we just check the first box, the default won't necessarily be what they want to look at or the most relevant data.

That being said, this code is done and makes the UI more robust so after simplifying the message, let's merge it even if we do something else after! Thank you for these changes @MatthewKhouzam! 🍰 

Are there other reasons this solution was proposed instead of checking a box by default (ex. A message is significantly easier to implement)?

----

Justification for message change:
* We don't need to say "Trace analysis completed" since this is implied when we tell them they can view results 
* I don't think users would understand what "the tree" is. I only understood because I've seen the code. "Check a box" is more specific guidance.

File: packages/react-components/src/components/xy-output-component.tsx [hunk 2]
```diff
@@ -232,6 +232,16 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
                 }
             </React.Fragment>;
         }
+        if (this.state.outputStatus === ResponseStatus.COMPLETED && this.state.xyData?.datasets?.length===0 ) {
+            return <React.Fragment>
+                {
+                    <p className='no-selected' style={{fontSize: 20, marginRight: '5px', marginLeft: '5px', justifyContent:'center', alignItems:'center'}}>
+                        <p>Trace analysis completed.</p>
+                        <p>Select an element in the tree to visualize output.</p>
```
[the remark this answers is not in the document]
Comment (bhufmann) [reply packages/react-components/src/components/xy-output-component.tsx:240]: What about: "Select a checkbox to see analysis results"

File: packages/react-components/src/components/xy-output-component.tsx [hunk 3]
```diff
@@ -232,6 +232,16 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
                 }
             </React.Fragment>;
         }
+        if (this.state.outputStatus === ResponseStatus.COMPLETED && this.state.xyData?.datasets?.length===0 ) {
+            return <React.Fragment>
+                {
+                    <p className='no-selected' style={{fontSize: 20, marginRight: '5px', marginLeft: '5px', justifyContent:'center', alignItems:'center'}}>
```
Comment (bhufmann) [inline packages/react-components/src/components/xy-output-component.tsx:238]: I see the following error in the console:
```
root ERROR Warning: validateDOMNesting(...): <p> cannot appear as a descendant of <p>.
    in p (created by XYOutputComponent)
    in p (created by XYOutputComponent)
    in div (created by XYOutputComponent)
    in div (created by XYOutputComponent)
    in div (created by XYOutputComponent)
    in XYOutputComponent (created by TraceContextComponent)
    in Resizable (created by GridItem)
    in DraggableCore (created by GridItem)
    in GridItem (created by ReactGridLayout)
    in div (created by ReactGridLayout)
    in ReactGridLayout (created by ResponsiveReactGridLayout)
    in ResponsiveReactGridLayout (created by WidthProvider)
    in WidthProvider (created by TraceContextComponent)
    in div (created by TraceContextComponent)
    in TraceContextComponent
    in div
```

**Review by bhufmann — CHANGES_REQUESTED** [review]
Nice improvements. Please reference issue #406 and fix linting failure.

File: packages/react-components/src/components/xy-output-component.tsx (hunk 3 above)
Comment (MatthewKhouzam) [reply packages/react-components/src/components/xy-output-component.tsx:238]: Huh... I'll remove it then... it works though. ?!?

File: packages/react-components/src/components/xy-output-component.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (MatthewKhouzam) [reply packages/react-components/src/components/xy-output-component.tsx:240]: Picking Bernd's because I checked my email late.

**Comment by MatthewKhouzam** [comment]
Fixes #604. Updated commit message.

**Review by Rodrigoplp-work — APPROVED** [review]

**Review by bhufmann — APPROVED** [review]
Thanks. Looks good to me.

**Comment by MatthewKhouzam** [comment]
@ebugden Any thoughts on the changes?

**Review by ebugden — APPROVED** [review]
Looks good to me! Thank you for these changes @MatthewKhouzam!

**Review by bhufmann — APPROVED** [review]
It still looks good to me. Thanks!

**Review by Rodrigoplp-work — APPROVED** [review]

**Comment by MatthewKhouzam** [comment]
Thanks all!

**Pull request merged**